packages/go_router/lib/src/configuration.dart

```diff
@@ -0,0 +1,145 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'configuration.dart';
+import 'logging.dart';
+import 'path_utils.dart';
+import 'typedefs.dart';
+
+export 'route.dart';
+export 'state.dart';
+
+/// The route configuration for GoRouter configured by the app.
+class RouteConfiguration {
```

Comment:
The properties in this class are only needed in gorouteinformationparser once we remove the namedLocation. I am a bit unsure whether we should introduce a new class that will probably be deprecated in the future

Reply:
The routeConfiguration field is marked `@visibleForTesting` in RouterDelegate, so I don't consider this a new public API. The reason for making this a separate class is make it easier to change the configuration without updating all of the places where the routeConfiguration is being passed around. For example, we might want to deprecate `topRedirect`.

Reply:
but at somepoint we won't need to pass it around.

We would need to expose this class if we expose the GoRouterDelegate, otherwise, developer won't be able to use GoRouterDelegate constructor directly.

Reply:
Filed an issue to remove this along with the namedLocation API: https://github.com/flutter/flutter/issues/108139